src/mobu/business/jupyterloginloop.py

```diff
@@ -59,6 +59,7 @@ async def execute(self) -> None:
         await self.lab_settle()
         if self.stopping:
             return
+        await self.lab_login()
```

Comment:
Is it this that ensures we have the right token/cookies ?

Reply:
Yeah, I moved this from JupyterClient to the business so that we get logging and timing.